sds/src/proximity_keywords/mod.rs

```diff
@@ -279,19 +279,29 @@ fn get_char_type(c: &char) -> CharType {
 
 /// Function that standardizes a list of characters, by pushing characters one by one in a standard way.
 /// Takes a closure that will be called when a character is to be pushed
-pub fn standardize_path_chars<F>(chars: &str, mut push_character: F)
+pub fn standardize_path_chars<F>(characters: &str, mut push_character: F)
 where
     F: FnMut(&char),
 {
-    let kw_length = chars.len();
+    let kw_length = characters.len();
+    if kw_length == 0 {
+        return;
+    }
 
-    if let Some(c) = chars.chars().next() {
-        push_character(&c);
-    } else {
+    let mut char_iterator = characters.chars();
+    if char_iterator.all(|c| c.is_ascii_lowercase())
+        || char_iterator.all(|c| c.is_ascii_uppercase())
+    {
+        for c in characters.chars() {
+            push_character(&c)
```

Comment:
The method push_character can possibly be updated to push string slice. This should then be a bit faster

Reply:
If I'm reading it right, this is just a short-cut to prevent doing the full sanitization in some common cases? In that case this should happen before the String allocation in `sanitize()`. At a minimum this can just call `to_owned()` on the input string (so it allocates the exact amount needed, and this be more likely to use an efficient mem-copy implementation). Depending on how this sanitized path is used, you could consider returning a `Cow<str>` which will allow returning either the original input slice (no copy needed) or a `String` (if it needs to be modified).

Reply:
I've done this: 
[Add should_bypass_standardize fn](https://github.com/DataDog/dd-sensitive-data-scanner/pull/74/commits/4171144bca5889d43d76b98d693b4f85ec310b6e) and
[Use should_bypass in standardize_path](https://github.com/DataDog/dd-sensitive-data-scanner/pull/74/commits/5fb83ecdc9e031678da9a0d72f9f18f889a2ddc4)

Tell me what you think :)